### ISSUE TITLE
--recursive fail to parse folder with __init__.py at the root when using relative path

### ISSUE THREAD

**Issue by CareF** [pylint-dev/pylint#9210]
### Bug description

```python
# folder struct
./__init__.py
./a.py # same goes if a.py is in a subfolder

# __init__.py is empty

# a.py includes anything that may trigger an error report like:

def foo():
    unused = 1
```

### Command used

```shell
pylint --recursive=y ./
```

### Pylint output

```python
None
```

### Expected behavior

Should be like `pylint a.py` or `pylint --recursive=y $(pwd)`

```
************* Module testpy.a
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
a.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
a.py:2:4: W0612: Unused variable 'unused' (unused-variable)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.6 (main, Oct  2 2023, 20:46:17) [Clang 14.0.0 (clang-1400.0.29.202)]
```

### OS / Environment

MacOS

**Comment by CareF** [comment]
I can track it to L146 at `lint/expand_modules.py`
